### ISSUE TITLE
Windows - ModuleNotFoundError: No module named 'fcntl'

### ISSUE THREAD

**Issue by Oisins** [Bogdanp/dramatiq#75]
# Issues

GitHub issues are for bugs.  If you have questions, please ask them on the [discussion board](https://discuss.dramatiq.io).

## Checklist

* [ ] Does your title concisely summarize the problem?
* [ ] Did you include a minimal, reproducible example?
* [ ] What OS are you using?
* [ ] What version of Dramatiq are you using?
* [ ] What did you do?
* [ ] What did you expect would happen?
* [ ] What happened?

## What OS are you using?
Windows 10 64bit

## What version of Dramatiq are you using?
1.1

When running dramatiq on windows I get the "ModuleNotFoundError: No module named 'fcntl'" error. Does dramatiq support windows? Are there any workarounds?

**Comment by ericgj** [comment]
I ran into this too. I worked around it by placing a blank `fcntl.py` in my project root dir.  

It might be nice if Prometheus were made an optional dependency to avoid this, or even better do the file locking for the exposition server in an OS-agnostic way.

**Comment by Oisins** [comment]
@ericgj Thanks for the idea, I'll try it.

**Comment by Bogdanp** [comment]
Unfortunately, I don't have a Windows machine to test changes on and Travis doesn't support Windows. It looks like in order to implement file locking on windows, we'd have to bring in `pywin32` as an optional dependency for windows builds and extract the file-locking stuff into a compatibility module of sorts. I'd gladly take PRs on this if anyone is willing to tackle it, but it's unlikely that I'll personally get to it any time soon.

**Comment by Bogdanp** [comment]
@Oisins go for it. Looking at its source code its only dependency is `pywin32` on Windows so that's good.

**Comment by ericgj** [comment]
Yeah, it's in `__main__`, i.e. the dramatiq executable. On Windows I basically worked around it by writing a [separate top-level script][worker] that runs a single worker process with multiple threads, rather than managing multiple forked processes each with multiple threads. Presumably then you could run several instances of such a script if you need multiple worker processes, for my needs I've found one process is fine. 

I also [set it up to run as a Windows service][service], with much trial and error. Have not gotten signals to work right though (a common issue on Windows) and right now it forces workers to terminate.

Hope this helps...  

I think changing `__main__` to work cross-platform might be too nasty (hard enough to reason about code with os.fork() in the middle of it), but including a separate executable script for Windows, e.g. `dramatiq_win.cmd` might be worth it. 

[worker]: https://bitbucket.org/ericgj/ert-cdp-task-queue/src/a079bd1b83d8d058c867c5aaf4a77f7e6786684c/bin/worker.py?at=master&fileviewer=file-view-default

[service]: https://bitbucket.org/ericgj/ert-cdp-task-queue/src/a079bd1b83d8d058c867c5aaf4a77f7e6786684c/bin/worker_service.py?at=master&fileviewer=file-view-default

**Comment by Bogdanp** [comment]
I think we should take Eric's approach for now by fixing the `fcntl` issue and adding docs on how to run your own `Worker`.  I think we'll also have to remove `TimeLimit` from the list of default middleware on Windows since it relies on `signal.setitimer`.
